### PR TITLE
Products: Improving loading of product dropdown.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

While testing other PRs, I noticed the products dropdown was a little jumpy, this PR makes the loading process a little cleaner and stops it from jumping in. 

This error helps, the API call fails or something goes wrong, the dropdown is somewhat usable. 

#### Before

https://user-images.githubusercontent.com/237508/170468600-228e4b8e-486a-46b0-a127-adcc2314244e.mov

#### After

https://user-images.githubusercontent.com/237508/170468193-63a11c9a-28c4-45f8-bcc1-828dc98519ec.mov

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by timarney** [comment]
👍 

Would be nice to test this with the currently skipped Karma test

```javascript
xdescribe('Product floating menu', () => {
```

**Comment by spacedmonkey** [comment]
> Would be nice to test this with the currently skipped Karma test

Re-enabling these tests, they still fail. 
Screenshot 2022-05-26 at 12 16 30

**Review by timarney — APPROVED** [review]
Nice

File: packages/story-editor/src/components/library/panes/shopping/productDropdown.js [hunk 1]
```diff
@@ -34,8 +39,12 @@ const StyledDropDown = styled(Datalist.DropDown)`
 function ProductDropdown({ product, setProduct }) {
   const isShoppingIntegrationEnabled = useFeature('shoppingIntegration');
 
+  const initialProducts = useMemo(
```
Comment (swissspidy) [inline packages/story-editor/src/components/library/panes/shopping/productDropdown.js:42]: FWIW using `useMemo` for this is overkill

**Pull request merged**